panic: runtime error: index out of range

I'm unsure what I did to break the logger, but it's broken and in a "CrashLoopBackOff" and I don't know how to recover it.

This occurred while I was in a long session of `deis pull`ing my app as a Docker image with small changes to the code each time.

I haven't looked into this Go stacktrace yet. Any interpretations of this?

``` sh
[chexxor@fedora myproject]$ kubectl --namespace=deis logs deis-logger-9343270-5l8d8
2017/01/31 04:17:15 INF    1 [logs/consume] (10.71.240.167:4150) connecting to nsqd
2017/01/31 04:17:15 Log aggregator running
2017/01/31 04:17:15 Weblog server serving at http://[::]:8088
panic: runtime error: index out of range

goroutine 49 [running]:
panic(0x7afc00, 0xc42000c0a0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/deis/logger/log.buildApplicationLogMessage(0xc420083540, 0xc420315c50, 0x816c4a)
	/go/src/github.com/deis/logger/log/message_handler.go:61 +0x3b8
github.com/deis/logger/log.handle(0xc42013001e, 0x1d0, 0x1d0, 0x9f20a0, 0xc42010c780, 0xc420013130, 0xc4201184d8)
	/go/src/github.com/deis/logger/log/message_handler.go:33 +0xf3
github.com/deis/logger/log.newNSQAggregator.func1(0xc420119c20, 0xc4202d0c01, 0x101)
	/go/src/github.com/deis/logger/log/nsq_aggregator.go:23 +0x57
github.com/deis/logger/vendor/github.com/nsqio/go-nsq.HandlerFunc.HandleMessage(0xc4201166c0, 0xc420119c20, 0x7b43a0, 0xc4201166c0)
	/go/src/github.com/deis/logger/vendor/github.com/nsqio/go-nsq/consumer.go:42 +0x30
github.com/deis/logger/vendor/github.com/nsqio/go-nsq.(*Consumer).handlerLoop(0xc420144000, 0x9ecac0, 0xc4201166c0)
	/go/src/github.com/deis/logger/vendor/github.com/nsqio/go-nsq/consumer.go:1105 +0x14a
created by github.com/deis/logger/vendor/github.com/nsqio/go-nsq.(*Consumer).AddConcurrentHandlers
	/go/src/github.com/deis/logger/vendor/github.com/nsqio/go-nsq/consumer.go:1087 +0xaa
```

--- a/log/message_handler.go
+++ b/log/message_handler.go
@@ -53,11 +53,17 @@ func buildControllerLogMessage(message *Message) string {
 }
 
 func buildApplicationLogMessage(message *Message) string {
-	return fmt.Sprintf("%s %s[%s.%s.%s]: %s",
+	p := podRegex.FindStringSubmatch(message.Kubernetes.PodName)
+	tag := fmt.Sprintf(
+		"%s.%s",
+		message.Kubernetes.Labels["type"],
+		message.Kubernetes.Labels["version"])
+	if len(p) > 0 {
+		tag = fmt.Sprintf("%s.%s", tag, p[len(p)-1])
+	}
+	return fmt.Sprintf("%s %s[%s]: %s",
 		message.Time.Format(timeFormat),
 		message.Kubernetes.Labels["app"],
-		message.Kubernetes.Labels["type"],
-		message.Kubernetes.Labels["version"],
-		podRegex.FindStringSubmatch(message.Kubernetes.PodName)[4],
+		tag,
 		message.Log)
 }

--- a/log/message_handler_test.go
+++ b/log/message_handler_test.go
@@ -18,6 +18,9 @@ var (
 	validAppMessage = `{"log": "test message", "stream": "stderr", "time": "2016-10-18T20:29:38+00:00", "docker": {"container_id": "containerId"}, "kubernetes": {"namespace_name": "foo", "pod_id": "podId", "pod_name": "foo-web-845861952-nzf60", "container_name": "foo-web", "labels": {"app": "foo",
 "heritage": "deis", "type": "web", "version": "v2"}, "host": "host"}}`
 
+	badPodNameMessage = `{"log": "test message", "stream": "stderr", "time": "2016-10-18T20:29:38+00:00", "docker": {"container_id": "containerId"}, "kubernetes": {"namespace_name": "foo", "pod_id": "podId", "pod_name": "foo-web-845861952", "container_name": "foo-web", "labels": {"app": "foo",
+"heritage": "deis", "type": "web", "version": "v2"}, "host": "host"}}`
+
 	badjson = `{"log":}`
 )
 
@@ -60,6 +63,16 @@ func TestBuildApplicationLogMessageFromValidMessage(t *testing.T) {
 	expected := buildApplicationLogMessage(message)
 	assert.Equal(t, expected,
 		"2016-10-18T20:29:38+00:00 foo[web.v2.nzf60]: test message",
+		"failed to build application log")
+}
+
+func TestBuildApplicationLogMessageFromInvalidMessage(t *testing.T) {
+	message := new(Message)
+	err := json.Unmarshal([]byte(badPodNameMessage), message)
+	assert.NoError(t, err, "error occured parsing log message")
+	expected := buildApplicationLogMessage(message)
+	assert.Equal(t, expected,
+		"2016-10-18T20:29:38+00:00 foo[web.v2]: test message",
 		"failed to build application log")
 }
 

